l2geth/accounts/abi/abi.go

```diff
@@ -192,3 +194,25 @@ func (abi *ABI) EventByID(topic common.Hash) (*Event, error) {
 	}
 	return nil, fmt.Errorf("no event with id: %#x", topic.Hex())
 }
+
+// RevertSelector is a special function selector for revert reason unpacking.
+var RevertSelector = crypto.Keccak256([]byte("Error(string)"))[:4]
+
+// UnpackRevert resolves the abi-encoded revert reason. According to the solidity
+// spec https://solidity.readthedocs.io/en/latest/control-structures.html#revert,
```

Comment:
Best to pin it to a version and use the new solidity docs website:

```suggestion
// docs https://docs.soliditylang.org/en/v0.8.4/control-structures.html#revert,
```